modules/order/src/OrderStorage.php

```diff
@@ -83,10 +85,33 @@ public function loadUnchanged($id) {
   /**
    * {@inheritdoc}
    */
+  protected function doPreSave(EntityInterface $entity) {
+    $id = parent::doPreSave($entity);
+    /** @var \Drupal\commerce_order\Entity\OrderInterface $entity */
+    if ($entity->getRefreshState() == OrderInterface::REFRESH_ON_SAVE) {
+      $this->orderRefresh->refresh($entity);
+    }
+    // Only the REFRESH_ON_LOAD state needs to be persisted on the entity.
+    if ($entity->getRefreshState() != OrderInterface::REFRESH_ON_LOAD) {
+      $entity->setRefreshState(NULL);
+    }
+
+    return $id;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
   protected function postLoad(array &$entities) {
-    foreach ($entities as $entity) {
-      if (!$this->skipRefresh && $this->orderRefresh->needsRefresh($entity)) {
-        $this->orderRefresh->refresh($entity);
+    if (!$this->skipRefresh) {
+      /** @var \Drupal\commerce_order\Entity\OrderInterface[] $entities */
+      foreach ($entities as $entity) {
+        $explicitly_requested = $entity->getRefreshState() == OrderInterface::REFRESH_ON_LOAD;
+        if ($explicitly_requested || $this->orderRefresh->shouldRefresh($entity)) {
```

Comment:
I like this a lot. We'll only refresh on load if it's been a day or more. I'd expect most transactions to complete in under 5 minutes anyways.